### PR TITLE
JAMES-2703 Apache James post 3.3.0 Deprecation and removals

### PR DESCRIPTION
**Pull request by chibenwa**
Email thread: https://www.mail-archive.com/server-dev@james.apache.org/msg60016.html

### CODE REVIEW

File: mailbox/caching/pom.xml [hunk 1]
```diff
@@ -31,6 +31,8 @@
     <packaging>bundle</packaging>
     <name>Apache James :: Mailbox :: Caching</name>
 
+    <description>JAMES-2703 This maven module is deprecated and will be removed straight after upcoming James 3.4.0 release, unless it finds a maintainer</description>
+
```
Comment (rouazana) [inline mailbox/caching/pom.xml]: no explanation of why and what should be used instead?

File: server/data/data-jdbc/src/main/java/org/apache/james/user/jdbc/AbstractJdbcUsersRepository.java [hunk 2]
```diff
@@ -91,6 +91,10 @@
  * <td>Select a user by name (case-insensitive lowercase).</td>
  * </tr>
  * </table>
+ *
+ * @deprecated JAMES-2703 This class is deprecated and will be removed straight after upcoming James 3.4.0 release, unless it finds a maintainer
+ *
+ * Please migrate to other UsersRepository implementations
```
Comment (rouazana) [inline server/data/data-jdbc/src/main/java/org/apache/james/user/jdbc/AbstractJdbcUsersRepository.java]: maybe you can advertise for JpaUsersRepository?

**Comment by Arsnael** [comment]
```
[c41aa1bce9b69eff9337f608bd5b0ae04957fb5c] 7 MB | 0.4/1.1 MB[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:2.17:check (default) on project apache-james-mailbox-caching: You have 1 Checkstyle violation. -> [Help 1]
```

File: mailbox/caching/pom.xml (hunk 1 above)
Comment (rouazana) [reply mailbox/caching/pom.xml]: sorry, but can you add this commit in the java classes you deprecate also?

Comment (chibenwa) [reply mailbox/caching/pom.xml]: Done

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**